common/vm_get_id.yml

```diff
@@ -16,6 +16,6 @@
 
 - name: "Set fact of the VM ID of VM '{{ vm_name }}'"
   set_fact:
-    vm_id: "{{ vm_guest_facts.instance.moid | default(0) }}"
+    vm_id: "{{ vm_guest_facts.instance.moid.split('-')[-1] | int }}"
```

Comment:
If there is no "when" clause to check these properties are all defined, then need to add a default value for `vm_id` to avoid such referred property not defined issue, or add a check if "vm_id" got is "0", then the testing will fail due to not get the valid "vm_id".